source/js/components/join/language-select.jsx

```diff
@@ -22,11 +22,14 @@ export default class LanguageSelect extends React.Component {
 
     let lang_codes = Object.keys(languages);
     let options = lang_codes.map(lang_code => {
-      return (
-        <option key={lang_code} value={lang_code}>
-          {languages[lang_code]}
-        </option>
-      );
+      const optionAttributes = {
+        key: lang_code,
+        value: lang_code
+      };
+      if (this.props.selectedLang === lang_code) {
+        optionsAttributes.selected = "selected";
+      }
+      return <option {...optionAttributes}>{languages[lang_code]}</option>;
```

Comment:
this is where the value selection should be recorded, not on the `<select>` element.

Reply:
Good catch!